File: src/modelo/pet.ts
export default class Pet {
    private nome: string
    private tipo: string
    private raca: string
    private tamanho: string
    private genero: string

    constructor(nome: string, tipo: string, raca: string, genero: string, tamanho: string) {
        this.nome = nome
        this.tipo = tipo
        this.raca = raca
        this.genero = genero
        this.tamanho = tamanho
    }

    public get getNome() { return this.nome }
    public set setNome(nome: string) { this.nome = nome }

    public get getTipo() { return this.tipo }
    public get getRaca() { return this.raca }
    public get getGenero() { return this.genero }

    public get getTamanho() { return this.tamanho }
    public set setTamanho(tamanho: string) { this.tamanho = tamanho }

}

File: back/pl/src/main/java/com/fatec/pl/modelo/Pet.java
package com.fatec.pl.modelo;

import org.springframework.hateoas.RepresentationModel;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class Pet extends RepresentationModel<Pet> {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column
	private String nome;
	
	@Column
	private String tipo;
	
	@Column
	private String raca;
	
	@Column
	private String tamanho;
	
	@Column
	private String genero;
	
	@OneToOne(orphanRemoval = true, cascade = CascadeType.ALL)
	private Cliente dono;
}
